AzureData/Source/AzureDataExtensions.swift

```diff
@@ -74,9 +74,13 @@ public extension Database {
     
     //delete
     public func delete (_ user: User, callback: @escaping (DataResponse) -> ()) {
-        return DocumentClient.default.delete (user, callback: callback)
+        return self.delete(userWithId: user.id, callback: callback)
```

Comment:
rather than calling `self.delete` and extracting the `id`, just use the generic `DocumentClient.default.delete(Resource)`